### ISSUE TITLE
`reportInvalidTypeVarUse` false positive on constrained `TypeVar`s in protocols

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#898]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYA0UAKogKYBqAhiALABQXXtUAvHUasQACgBEtcTQDOMEDQBGcGAxkBKHp1IUWMmVABiYMKKJgS5CgG1aAXXUAuKAGIoDEOBDP6CBlAA3NiQWRQp-SXEoAFcZBgATTBQoNAYUDyRSQmIySihxYzAomQALMGiKRMV-ciCQEJQYLigWqHiGYECWCmiGUTiKYHUoAFoAPjpnADoZoA)

```python
from typing import Protocol, TypeVar


T = TypeVar("T", str, bytes)

class Foo(Protocol[T]): # error: Type variable "T" used in generic Protocol "Foo" should be covariant
    def value(self) -> T: ...
```

related: #893 #897

**Comment by KotlinIsland** [comment]
this is p1 imo
